### PR TITLE
Correcting Collection of Diagnostics from SI Tests

### PR DESCRIPTION
**Pull request by kensipe**
Summary:
correcting and being consistent without si tests around test result collection

JIRA issues:

### CODE REVIEW

**Comment by kensipe** [comment]
finally green tests!!!  
https://jenkins.mesosphere.com/service/jenkins/job/system-integration-tests/job/metronome-si-dcos-open/job/ken%252Ffix-diagnostic/2/

**Comment by kensipe** [comment]
For clarity example failure https://jenkins.mesosphere.com/service/jenkins/job/system-integration-tests/job/metronome-si-dcos-open/job/master/529/console
```ERROR: No artifacts found that match the file pattern "diagnostics.zip". Configuration error?
Finished: FAILURE```

this PR resolves this issue.

**Comment by kensipe** [comment]
ugh
```
bin/install-protobuf.sh: 10: [: Linux: unexpected operator
E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)
E: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?
```

File: Jenkinsfile.open.si [hunk 1]
```diff
@@ -37,7 +37,8 @@ node('py36') {
                 tokenCredentialId: "4a6636e5-452e-41d2-aea3-9f28448196f3")
         } finally {
             junit allowEmptyResults: true, testResults: "**/shakedown.xml"
-            archiveArtifacts "diagnostics.zip"
+            archive includes: "**/diagnostics.zip"
```
Comment (takirala) [inline Jenkinsfile.open.si]: From my observations, i have found `archive` step to be not working. See my comment https://github.com/dcos/metronome/pull/341#discussion_r294846776 where it worked after changing `archive` (which is deprecated) to `archiveArtifacts`.

You can allow "empty" archives by adding `, allowEmptyArchive: true` [like here](https://github.com/dcos/metronome/pull/341/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R30)

Either way, if you know that `archive` would work, this lgtm.

**Pull request merged**